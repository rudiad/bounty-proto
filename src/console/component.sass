.console
  bottom: 0
  position: absolute
  border: 1px solid black
  padding: .7em
  left: 0
  right: 0
  margin-left: auto
  margin-right: auto
  height: 10em
  color: white
  background-color: #241d30
  opacity: 0.8
  overflow-y: auto
  font-family: "Lucida Console", Monaco, monospace
  z-index: -1
