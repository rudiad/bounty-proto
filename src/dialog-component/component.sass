$top-bar-dimension: 40px

.dialog__backdrop
  position: fixed
  top: 0
  bottom: 0
  left: 0
  right: 0
  background-color: #19171C
  opacity: 0.75
  padding: 50
  height: 100vh

.dialog
  background-color: white
  border-radius: 5
  width: 66%
  margin: 0 auto
  border: 1px solid #DAD8DE
  position: absolute
  top: 20%
  left: 20%

.container
  display: flex

.container__block
  display: inline-block

.container__left-block
  @extend .container__block
  float: left
  margin-left: 20px

.container__right-block
  @extend .container__block
  float: right

.top-bar
  @extend .container
  height: $top-bar-dimension
  justify-content: space-between

.top-bar-element
  width: auto
  text-align: center
  vertical-align: middle
  line-height: $top-bar-dimension
  font-weight: lighter

.top-bar-element-left
  @extend .container__left-block
  @extend .top-bar-element

.top-bar-element-right
  @extend .container__right-block
  @extend .top-bar-element

.dialog__divider-width
  width: 96%

@mixin bottom-bar-box($color, $bg-color)
  color: $color
  background-color: $bg-color
  border: 1px solid #7D55C7
  float: left
  text-align: center
  width: 80px
  height: 30px
  vertical-align: middle
  line-height: 30px
  cursor: pointer
  font-size: 16px
  margin-left: 20px
  margin-top: 10px
  margin-bottom: 30px

.title
  font-size: 13px
  font-weight: lighter
