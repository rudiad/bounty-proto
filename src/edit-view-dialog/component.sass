@import "../dialog-component/component.sass"

.edit-view__background
  @extend .dialog__backdrop

.edit-view__dialog-component
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__dialog-mobile
  @extend .dialog
  height: 275px
  width: 415px
  left: 35%

.edit-view__dialog-overlay
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__content
  @extend .container__left-block
  overflow-y: auto
  height: 50%
  width: 95%
  display: flex
  flex-direction: column

.edit-subcontainer__input
  margin-top: .5em
  margin-bottom: .5em

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
  margin-left: 10px
