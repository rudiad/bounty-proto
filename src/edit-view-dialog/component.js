import React, { Component } from 'react';
import PropTypes from 'prop-types';
import closeButton from '../img/close_icon.png';
import './component.sass';
import { RadioOption } from '../extension-view-dialog/radio-option';
import { DefaultMobileOrientation, MobileSizes } from '../constants/mobile';
import { OverlaySizes } from '../constants/overlay-sizes';
import { ViewerTypes } from '../constants/viewer-types';
import { IdentityOptions } from '../constants/identity-options';
import { MobileOrientation, MobileScreenSizes} from '../constants/mobile';
const { ExtensionViewType } = window['extension-coordinator'];

export class EditViewDialog extends Component {
  constructor(args) {
    super(args);
    this.state = {
      x: 0,
      y: 0,
      orientation: DefaultMobileOrientation,
      frameSize: {
        width: 0,
        height: 0,
      },
    }
  }

  renderFrameSizeComponents(sizeSet) {
    return Object.keys(sizeSet).map(key => {
      return <RadioOption key={key} name="frameSize" value={key} onChange={this.onChange} checked={key === this.state.frameSize}/>
    });
  }

  renderViewerTypeComponents() {
    return Object.keys(ViewerTypes).map(key => {
      return <RadioOption key={key} name="viewerType" value={ViewerTypes[key]} onChange={this.onChange} checked={ViewerTypes[key] === this.state.viewerType}/>
    });
  }

  renderIdentityOptionComponents() {
    return Object.keys(IdentityOptions).map(option => {
      return <RadioOption key={option} name="identityOption" value={option} onChange={this.onChange} checked={option === this.state.identityOption}/>
    });
  }

  renderOrientationComponents() {
    return Object.keys(MobileOrientation).map(option => {
      return <RadioOption key={option} name="orientation" value={MobileOrientation[option]} onChange={this.onChange} checked={option === this.state.orientation}/>
    });
  }
  onChange = (input) => {
    const newState = {};
    newState[input.target.name] = input.target.value;
    this.setState(newState);
  }

  componentDidMount() {
    this.props.views.forEach(element => {
      if (element.id === this.props.idToEdit) {
        this.setState({
          x: element.x,
          y: element.y,
          orientation: element.orientation,
          frameSize: element.frameSize,
          type: element.type
        });
      }
    });
  }

  editClassFromType() {
    let editClass;
    switch (this.state.type) {
      case ExtensionViewType.Component:
        editClass = 'edit-view__dialog-component';
        break;
      case ExtensionViewType.Mobile:
        editClass = 'edit-view__dialog-mobile';
        break;
      case ExtensionViewType.Overlay:
        editClass = 'edit-view__dialog-overlay';
        break;
      default:
        editClass = 'edit-view__dialog'
    }
    return editClass;
  }

  render() {
    if (!this.props.show) {
      return null;
    }

    return (
      <div className='edit-view'>
        <div className="edit-view__background"/>
        <div className={this.editClassFromType()}>
          <div className="dialog__top-bar-container">
            <div className="top-bar-container__title"> Edit View </div>
            <div className="top-bar-container__escape" onClick={this.props.closeHandler}><img alt="Close" src={closeButton}/></div>
          </div>
          <hr className="dialog__divider" />

          {this.state.type === ExtensionViewType.Component &&
            <div className="edit-view__content">
              <div className="type-and-size-container__type-title">
                Position (%)
              </div>
              <div className='overlay-custom-container'>
                <div className='overlay-custom-container'>
                  <div className="edit-subcontainer__input">
                    <label className="inputs__option-label">X</label>
                    <input type="text" name="x" value={this.state.x} onChange={this.onChange} />
                  </div>
                  <div className="edit-subcontainer__input">
                    <label className="inputs__option-label">Y</label>
                    <input type="text" name="y" value={this.state.y} onChange={this.onChange} />
                  </div>
                </div>
              </div>
            </div>}

            {this.state.type === ExtensionViewType.Overlay &&
            <div className="edit-view__content">
              <div className="type-and-size-container__type-title">
                Position (%)
              </div>
              <div className='overlay-custom-container'>
                <div className='overlay-custom-container'>
                  <div className="edit-subcontainer__input">
                    <label className="inputs__option-label">X</label>
                    <input type="text" name="x" value={this.state.x} onChange={this.onChange} />
                  </div>
                  <div className="edit-subcontainer__input">
                    <label className="inputs__option-label">Y</label>
                    <input type="text" name="y" value={this.state.y} onChange={this.onChange} />
                  </div>
                </div>
              </div>
            </div>}

            {this.state.type === ExtensionViewType.Mobile &&
            <div className="size-title__size-subcontainer">
            <div className="size-subcontainer__presets">
              <div className="type-and-size-container__type-title">
                Screen Size
              </div>
              <div className='size-subcontainer__presets-container'>
                <div className='size-subcontainer__size-selection'>
                  {this.renderFrameSizeComponents(MobileSizes)}
                </div>
                <div className='overlay-custom-container'>
                    <RadioOption className='overlay-custom' name="frameSize" value="Custom" onChange={this.onChange} checked={"Custom" === ''} />
                  <div className='overlay-custom-container'>
                    <div className="custom-subcontainer__input">
                      <label className="inputs__option-label inputs__width-offset"> Width </label>
                      <input type="text" name="width" onChange={this.onChange}/>
                    </div>
                    <div className="custom-subcontainer__input">
                      <label className="inputs__option-label"> Height </label>
                      <input type="text" name="height" onChange={this.onChange}/>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="size-subcontainer__presets">
              <div className="type-and-size-container__type-title">
                Orientation
              </div>
              <div className='overlay-custom-container'>
                {this.renderOrientationComponents()}
              </div>
            </div>
          </div>}

          <hr className="dialog__divider"/>
          <div className="dialog_bottom-bar">
            <div className="bottom-bar__save" onClick={() => this.props.saveViewHandler(this.state)}> Save </div>
            <div className="bottom-bar__cancel" onClick={this.props.closeHandler}> Cancel </div>
          </div>
        </div>
      </div>
    );
  }
}

EditViewDialog.propTypes = {
  show: PropTypes.bool.isRequired,
  idToEdit: PropTypes.string.isRequired,
  views: PropTypes.array.isRequired,
  closeHandler: PropTypes.func.isRequired,
  saveViewHandler: PropTypes.func.isRequired,
};
