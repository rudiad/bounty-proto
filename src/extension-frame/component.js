import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

const { ExtensionPlatform, ExtensionViewType} = window['extension-coordinator'];

const IFRAME_CLASS = 'extension-frame';
const EXTENSION_FRAME_INIT_ACTION = 'extension-frame-init';

export class ExtensionFrame extends Component {
  componentDidMount() {
    if (this.iframe) {
      this.iframe.onload = this._extensionFrameInit;
    }
  }

  render() {
    return (
      <iframe
      ref={this._bindIframeRef}
        src={process.env.PUBLIC_URL + '/extension-frame.html'}
        frameBorder={0}
        className={'rig-frame ' + IFRAME_CLASS}
        title={this.props.frameId}/>
    );
  }

  _bindIframeRef = (iframe) => {
    this.iframe = iframe;
  }

  _extensionFrameInit = () => {
    const extension = {
      anchor: this.props.type,
      channelId: this.props.extension.channelId,
      loginId: null,
      extension: this.props.extension,
      mode: this.props.mode,
      platform: (this.props.type === ExtensionViewType.Mobile) ? ExtensionPlatform.Mobile : ExtensionPlatform.Web,
      trackingProperties: {},
      iframeClassName: IFRAME_CLASS,
    }
    const data = {
      extension: extension,
      action: EXTENSION_FRAME_INIT_ACTION,
      frameId: this.props.frameId,
    }
    this.iframe.contentWindow.postMessage(data, '*');
  }

  _onIdentityLinked(isLinked) {}

  _onFrameDoubleClick(evt) {
    evt.preventDefault();
  }

  _onModalRequested(confirmationRequest) {}
}

ExtensionFrame.propTypes = {
  className: PropTypes.string.isRequired,
  frameId: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
};
