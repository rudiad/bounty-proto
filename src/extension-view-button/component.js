import React, { Component } from 'react';
import PropTypes from 'prop-types';
import line from '../img/rig_line.png';
import plus from '../img/plus_icon.png';
import './component.sass'

export class ExtensionViewButton extends Component {
  render() {
    return (
      <div className='button-wrapper'>
        <img alt='line' className='background' src={line} />
        <div className="circle-button" onClick={this.props.onClick}>
          <img alt='Add new view' src={plus} />
        </div>
      </div>
    )
  }
}

ExtensionViewButton.propTypes = {
  onClick: PropTypes.func.isRequired
}
