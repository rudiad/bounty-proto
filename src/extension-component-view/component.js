import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ExtensionFrame } from '../extension-frame';
const { ExtensionMode, ExtensionViewType, getComponentPositionFromView, getComponentSizeFromView } = window['extension-coordinator'];

export class ExtensionComponentView extends Component {
  computeViewStyles() {
    const extension = this.props.extension;
    const positionFromView = getComponentPositionFromView(
      this.props.frameSize.width,
      this.props.frameSize.height,
      {
        x: this.props.position.x * 100,
        y: this.props.position.y * 100,
      });
    const sizeFromView = getComponentSizeFromView(
      this.props.frameSize.width,
      this.props.frameSize.height,
      extension.views.component);

    let viewStyles = {
      border: '1px solid #7D55C7',
      position: 'absolute',
      left: positionFromView.x + 'px',
      top: positionFromView.y + 'px',
      width: `${sizeFromView.width}px`,
      height: `${sizeFromView.height}px`,
    }

    if (extension.views.component.zoom) {
      viewStyles = {
        width: `${sizeFromView.width / sizeFromView.zoomScale}px`,
        height: `${sizeFromView.height / sizeFromView.zoomScale}px`,
        transformOrigin: '0 0',
        transform: `scale(${sizeFromView.zoomScale})`,
      }
    }

    return viewStyles;
  }

  render() {
    return (
      <div
        className="view component-view"
        style={{
          width: this.props.frameSize.width + 'px',
          height: this.props.frameSize.height + 'px',
        }}>
          <div style={this.computeViewStyles()}>
          <ExtensionFrame
            className="view"
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Component}
            mode={ExtensionMode.Viewer}
          />
        </div>
      </div>
    );
  }
}

ExtensionComponentView.propTypes = {
  id: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  frameSize: PropTypes.object.isRequired,
  position: PropTypes.object.isRequired,
  role: PropTypes.string,
};
