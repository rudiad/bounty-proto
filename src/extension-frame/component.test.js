import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { setupMountTest } from '../tests/enzyme-util/mount';
import { ExtensionForTest } from '../tests/constants/extension';
import { ExtensionFrame } from './component';
const { ExtensionViewType, ExtensionAnchor, ExtensionMode } = window['extension-coordinator'];

describe('<ExtensionFrame />', () => {
  const setupShallow = setupShallowTest(ExtensionFrame, () => ({
    className: 'view',
    frameId: '0',
    extension: ExtensionForTest,
    type: ExtensionAnchor.Panel,
    mode: ExtensionMode.Viewer,
  }));

  const setupMount = setupMountTest(ExtensionFrame, () => ({
    className: 'view',
    frameId: '0',
    extension: ExtensionForTest,
    type: ExtensionAnchor.Panel,
    mode: ExtensionMode.Viewer,
  }));

  it('prevents the default when double clicked', () => {
    const mockEvent = {};
    mockEvent.preventDefault = jest.fn();
    const { wrapper } = setupShallow();
    wrapper.instance()._onFrameDoubleClick(mockEvent);
    expect(mockEvent.preventDefault).toHaveBeenCalled();
  });

  it('onload postMessages data correctly', () => {
    const { wrapper } = setupMount();

    const mockIframeRef = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };

    wrapper.instance().iframe = mockIframeRef
    wrapper.instance()._extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith({
      "action": "extension-frame-init",
      "extension": {
        "anchor": "panel",
        "channelId": "channelId",
        "extension": {
          "authorName": "test",
          "channelId": "channelId",
          "description": "description",
          "iconUrl": "icon_url",
          "id": "id",
          "name": "name",
          "requestIdentity": false,
          "sku": "sku",
          "state": "state",
          "summary": "summary",
          "token": "token",
          "vendorCode": "vendorCode",
          "version": "0.1",
          "views": {
            "component": { "aspectHeight": 3000, "aspectWidth": 2500, "viewerUrl": "test", "zoom": false },
            "config": { "viewerUrl": "test" },
            "liveConfig": { "viewerUrl": "test" },
            "panel": { "viewerUrl": "test" }
          },
          "whitelistedConfigUrls": ["foo"],
          "whitelistedPanelUrls": ["bar"] },
          "iframeClassName": "extension-frame",
          "loginId": null,
          "mode": "viewer",
          "platform": "web",
          "trackingProperties": {}
        },
      "frameId": "0"
    }, "*");
  });

  it('onload postMessages data correctly when platform is mobile', () => {
    const { wrapper } = setupMount({
      type: ExtensionViewType.Mobile
    });

    const mockIframeRef = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };

    wrapper.instance().iframe = mockIframeRef
    wrapper.instance()._extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith({
      "action": "extension-frame-init",
      "extension": {
        "anchor": "mobile",
        "channelId": "channelId",
        "extension": {
          "authorName": "test",
          "channelId": "channelId",
          "description": "description",
          "iconUrl": "icon_url",
          "id": "id",
          "name": "name",
          "requestIdentity": false,
          "sku": "sku",
          "state": "state",
          "summary": "summary",
          "token": "token",
          "vendorCode": "vendorCode",
          "version": "0.1",
          "views": {
            "component": { "aspectHeight": 3000, "aspectWidth": 2500, "viewerUrl": "test", "zoom": false },
            "config": { "viewerUrl": "test" },
            "liveConfig": { "viewerUrl": "test" },
            "panel": { "viewerUrl": "test" }
          },
          "whitelistedConfigUrls": ["foo"],
          "whitelistedPanelUrls": ["bar"] },
          "iframeClassName": "extension-frame",
          "loginId": null,
          "mode": "viewer",
          "platform": "mobile",
          "trackingProperties": {}
        },
      "frameId": "0"
    }, "*");
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.LiveConfig,
        mode: ExtensionMode.Dashboard,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.LiveConfig,
        mode: ExtensionMode.Dashboard,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.Config,
        mode: ExtensionMode.Config,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in panel mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Panel,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Overlay,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Overlay,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
