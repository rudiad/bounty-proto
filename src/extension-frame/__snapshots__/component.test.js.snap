// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionFrame /> when in config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    className="view"
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    mode="config"
    type="config"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in live config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    className="view"
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in live config mode renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    className="view"
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in panel mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    className="view"
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    mode="viewer"
    type="panel"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in video overlay mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    className="view"
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    mode="viewer"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in video overlay mode renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    className="view"
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    mode="viewer"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
